### ISSUE TITLE
Create PHP RFC

### ISSUE THREAD

**Issue by Renkas** [crazyxman/simdjson_php#8]
Could you create RFC to maybe implement this into PHP? The speed imporovement is impressive and would improve PHP itself.

https://wiki.php.net/rfc

**Comment by kocsismate** [comment]
Hi @crazyxman,

I've just provided a PR to php-src in order to bundle your extension to the core: https://github.com/php/php-src/pull/6551 and started a discussion about it: https://externals.io/message/112638

**Comment by enumag** [comment]
In case it's not accepted into PHP would you consider making it available via PECL?

**Comment by kocsismate** [comment]
Yes, it's most probably won't be accepted just yet. But in this case, I'll try my best to make it available via PECL, sure.

**Comment by crazyxman** [comment]
@kocsismate  thank you very much！ It is a good idea to make it available through PECL

**Comment by sandrokeil** [comment]
@kocsismate I'm looking forward for your support to make it available via pecl and to improve this PHP extension.

**Comment by kocsismate** [comment]
In the light of the mailing list discussion (https://externals.io/message/112638), I abandon my proposal, as people would rather prefer to use libsimdjson as an optional JSON parsing backend.

**Comment by TysonAndre** [comment]
> People in the ML suggested to make simdjson available as a parsing backend for the built-in json-related functions

That seems viable once implemented and the test cases of edge cases pass - if simdjson fails to parse it, then json_decode() can fall back to the original json implementation in case of https://www.php.net/json_last_error failing.

**Comment by TysonAndre** [comment]
The pecl simdjson 3.0.0 release should be in a good state (including for zts releases)

This will speed up the happy path, where the json string is valid json (see previous comment for error handling).

- Edge cases for floating-point numbers should be consistent with the patched simdjson code
- ZTS should work properly. Edge cases such as unicode surrogate pairs are now also consistent for error handling.
- Depth handling when arrays do/don't have properties is off by one and will need to be handled in the fork of the C simdjson library
- simdjson should be avoided for `len > 4GB` or `depth > SOME_REASONABLE_C_STACK_DEPTH` (e.g. have to account for smaller stacks in https://www.php.net/fiber - bison does not use the C stack)

I'd still need to fix https://github.com/crazyxman/simdjson_php/pull/81
